src/Secure/OpenIdConnectAuthentication.php

```diff
@@ -95,8 +105,11 @@ protected function login(): void
     protected function validateToken(): array
     {
         $validationUrl    = "https://login.twinfield.com/auth/authentication/connect/accesstokenvalidation?token=";
+        $validationResult = false;
         try {
-            $validationResult = @file_get_contents($validationUrl . urlencode($this->accessToken));
+            if ($this->accessToken !== null) {
+                $validationResult = @file_get_contents($validationUrl . urlencode($this->accessToken));
```

Comment:
I don't think we need a try / catch here, but a check with `false`. Would you mind changing that? Maybe add the [http response header data](https://secure.php.net/manual/en/reserved.variables.httpresponseheader.php).

Reply:
I guess so. I cannot remember why I added the catch there. I did add a `@`, so it should not be able to throw exceptions.

Oh, and is that check: `if ($this->accessToken !== null)` really necessary?

Reply:
When I activated Charles to see what we were sending to Twinfield I noticed that we got a HTTP 400 Bad Request reply and that $accessToken was empty at all first calls. When we initiate the `OpenIdConnectAuthentication` class we provide a refresh token.

Call stack from what I remember: login -> **validateToken** -> refreshToken -> validateToken.

Reply:
Ah, I see. I think at the time I reasoned that the failed validation request was 'fine', as it then refreshes the token (as indicated by your call stack).

Perhaps it would be neater to add the following check to the `login` function instead, before the try-catch:
```
if ($this->accessToken === null) {
    $this->refreshToken();
}
```